src/ParticleDA.jl

```diff
@@ -180,35 +180,45 @@ of [`run_particle_filter`](@ref) to select the bootstrap filter.
 struct BootstrapFilter <: ParticleFilter end
 struct OptimalFilter <: ParticleFilter end
 
-# Get weights for particles by evaluating the probability of the observations predicted by the model
-# from independent normal pdfs for each observation.
-function get_log_weights!(weight::AbstractVector{T},
-                          obs::AbstractVector{T},
-                          obs_model::AbstractMatrix{T},
-                          weight_std::T) where T
 
-    nobs = size(obs_model,1)
-    @assert size(obs,1) == nobs
-
-    weight .= 1.0
-
-    for iobs = 1:nobs
-        weight .+= logpdf.(Normal(obs[iobs], weight_std), @view(obs_model[iobs,:]))
+# Get logarithm of unnormalized importance weights for particles for filter specific
+# proposal distribution
+function get_log_weights!(
+    log_weights::AbstractVector{T},  
+    observations::AbstractVector{T}, 
+    observation_means_given_particles::AbstractMatrix{T}, 
+    filter_data::NamedTuple,
+    filter_type::ParticleFilter,
+) where T
+    for p in 1:size(log_weights, 1)
+        log_weights[p] = compute_individual_particle_log_weight(
+            observations, 
+            observation_means_given_particles[:, p], 
+            filter_data, 
+            filter_type
+       )
     end
-
 end
 
-# Get weights for particles by evaluating the probability of the observations predicted by the model
-# from a multivariate normal pdf with mean equal to real observations and covariance equal to cov_obs
-function get_log_weights!(weight::AbstractVector{T},
-                          obs::AbstractVector{T},
-                          obs_model::AbstractMatrix{T},
-                          cov_obs::AbstractMatrix{T}) where T
-
-    weight .= Distributions.logpdf(Distributions.MvNormal(obs, cov_obs), obs_model)
-
+function compute_individual_particle_log_weight(
+    observations::AbstractVector{T},
+    observations_mean::AbstractVector{T},
+    filter_data::NamedTuple,
+    ::BootstrapFilter,
+) where T
+    difference = observations - observations_mean
+    return -0.5 * sum(difference.^2) / filter_data.obs_noise_std^2
```

Comment:
We have discussed this before but should we retain the difference between obs_noise_std and the previously defined filter_params.weight_std? Perhaps with more suitable naming?

Reply:
Ah good point yes we should probably sort this as part of this PR. I'd vote for just getting rid of `filter_params.weight_std` under the rationale in #205.

Reply:
I agree, I have just updated the PR with the removal of weight_std